tests/core/pyspec/eth2spec/test/merge/fork_choice/test_on_merge_block.py

```diff
@@ -0,0 +1,161 @@
+from eth2spec.utils.ssz.ssz_typing import uint256
+from eth2spec.test.context import spec_state_test, with_phases, MERGE
+from eth2spec.test.helpers.block import (
+    build_empty_block_for_next_slot
+)
+from eth2spec.test.helpers.fork_choice import (
+    get_genesis_forkchoice_store_and_block,
+    on_tick_and_append_step,
+    tick_and_add_block,
+)
+from eth2spec.test.helpers.state import (
+    state_transition_and_sign_block,
+)
+from eth2spec.test.helpers.fork_choice import (
+    prepare_empty_pow_block,
+)
+
+
+def with_pow_block_patch(spec, blocks, func):
+    def get_pow_block(hash: spec.Bytes32) -> spec.PowBlock:
+        for block in blocks:
+            if block.block_hash == hash:
+                return block
+        raise Exception("Block not found")
+    get_pow_block_backup = spec.get_pow_block
+    spec.get_pow_block = get_pow_block
+
+    class AtomicBoolean():
+        value = False
+    is_called = AtomicBoolean()
+
+    def wrap(flag: AtomicBoolean):
+        func()
+        flag.value = True
+
+    try:
+        wrap(is_called)
+    finally:
+        spec.get_pow_block = get_pow_block_backup
+    assert is_called.value
+
+
+@with_phases([MERGE])
+@spec_state_test
+def test_all_valid(spec, state):
```

Comment:
By default, the initial `state` is generated by [`initialize_beacon_state_from_eth1 `](https://github.com/ethereum/consensus-specs/blob/dev/specs/merge/beacon-chain.md#testing), where `state.latest_execution_payload_header` is filled. If we want a clean `state.latest_execution_payload_header`, you can do:
```suggestion
def test_all_valid(spec, state):
    state.latest_execution_payload_header = spec.ExecutionPayloadHeader()
```